go/controller/translator/adk_api_translator.go

```diff
@@ -0,0 +1,730 @@
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
+
+	outputs.ConfigMap = &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: objMeta,
+		Data:       map[string]string{},
+	}
+
+	if err := controllerutil.SetControllerReference(agent, outputs.Deployment, a.kube.Scheme()); err != nil {
+		return nil, err
+	}
+
+	if err := controllerutil.SetControllerReference(agent, outputs.Service, a.kube.Scheme()); err != nil {
+		return nil, err
+	}
+
+	if err := controllerutil.SetControllerReference(agent, outputs.ConfigMap, a.kube.Scheme()); err != nil {
+		return nil, err
+	}
+
+	return outputs, nil
+}
+
+func defaultDeploymentSpec(name string) *v1alpha1.DeploymentSpec {
+	// TODO: Come up with a better way to do tracing config for the agents
+	envVars := slices.Collect(utils.Map(
+		utils.Filter(
+			slices.Values(os.Environ()),
+			func(envVar string) bool {
+				return strings.HasPrefix(envVar, "OTEL_")
+			},
+		),
+		func(envVar string) corev1.EnvVar {
+			parts := strings.SplitN(envVar, "=", 2)
+			return corev1.EnvVar{
+				Name:  parts[0],
+				Value: parts[1],
+			}
+		},
+	))
+
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "KAGENT_NAMESPACE",
+		ValueFrom: &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: "metadata.namespace",
+			},
+		},
+	})
+
+	return &v1alpha1.DeploymentSpec{
+		Replicas: ptr.To(int32(1)),
+		Container: corev1.Container{
+			Name:            "kagent",
+			Image:           fmt.Sprintf("cr.kagent.dev/kagent-dev/kagent/app:%s", version.Get().Version),
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command:         []string{"uv", "run", "kagent", "static", "--host", "0.0.0.0", "--port", "8080", "--filepath", "/config/config.json"},
+			Ports: []corev1.ContainerPort{
+				{
+					Name:          "http",
+					ContainerPort: 8080,
+				},
+			},
+			Env: envVars,
+			ReadinessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Path: "/health",
+						Port: intstr.FromString("http"),
+					},
+				},
+				InitialDelaySeconds: 15,
+				PeriodSeconds:       3,
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "config",
+					MountPath: "/config",
+				},
+			},
+		},
+		Volumes: []corev1.Volume{
+			{
+				Name: "config",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: name,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (a *adkApiTranslator) translateDeclarativeAgent(ctx context.Context, agent *v1alpha1.Agent, state *tState) (*adk.AgentConfig, []corev1.EnvVar, error) {
+
+	model, envVars, err := a.translateModel(ctx, agent.Namespace, agent.Spec.ModelConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cfg := &adk.AgentConfig{
+		KagentUrl:   fmt.Sprintf("http://kagent.%s.svc.cluster.local:8083", common.GetResourceNamespace()),
```

Comment:
consider: remove `cluster.local` suffix for compatability with clusters not using this suffix? 
another option is to detect the suffix